Userland/Libraries/LibWeb/DOM/ShadowRoot.cpp

```diff
@@ -61,16 +61,35 @@ EventTarget* ShadowRoot::get_parent(Event const& event)
     return host();
 }
 
-// https://w3c.github.io/DOM-Parsing/#dom-innerhtml-innerhtml
+// https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#dom-shadowroot-innerhtml
 WebIDL::ExceptionOr<String> ShadowRoot::inner_html() const
 {
     return serialize_fragment(DOMParsing::RequireWellFormed::Yes);
 }
 
-// https://w3c.github.io/DOM-Parsing/#dom-innerhtml-innerhtml
-WebIDL::ExceptionOr<void> ShadowRoot::set_inner_html(StringView markup)
+// https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#dom-shadowroot-innerhtml
+WebIDL::ExceptionOr<void> ShadowRoot::set_inner_html(StringView value)
 {
-    TRY(DOMParsing::inner_html_setter(*this, markup));
+    // FIXME: 1. Let compliantString be the result of invoking the Get Trusted Type compliant string algorithm with TrustedHTML, this's relevant global object, the given value, "ShadowRoot innerHTML", and "script".
+
+    // 2. Let context be this's host.
+    auto context = this->host();
+
+    // 3. Let fragment be the result of invoking the fragment parsing algorithm steps with context and compliantString. FIXME: Use compliantString instead of markup.
+    auto fragment = TRY(verify_cast<Element>(*context).parse_fragment(value));
+
+    // 4. Replace all with fragment within this.
+    this->replace_all(fragment);
```

Comment:
This is different from the spec. This is due to a spec bug.

If I follow the spec `Replace all with fragment within context.` then some of the innerHTML tests in https://wpt.live/shadow-dom/ShadowRoot-interface.html fail.